TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift

```diff
@@ -268,10 +282,14 @@ import PromiseKit
                 if let linkedLogin = linkedLogin {
                     linkedBroker.error = nil
                     linkedBroker.linkedLogin = linkedLogin
-                    self.oAuthDelegate?.didLink?(linkedBroker: linkedBroker,
-                                                 userId: userId,
-                                                 userToken: userToken)
-                    onSuccess(linkedBroker)
+                    linkedBroker.authenticate(
+                        onSuccess: {
+                            self.oAuthDelegate?.didLink?(linkedBroker: linkedBroker, userId: userId, userToken: userToken)
+                            onSuccess(linkedBroker)
+                        },
+                        onSecurityQuestion: onSecurityQuestion,
+                        onFailure: onFailure
```

Comment:
Change `onFailure: onFailure` to `onFailure: onSuccess` because a failed authentication doesn't mean the linking failed.